tutorials/scripting/debug/debugger_panel.rst

```diff
@@ -39,29 +39,118 @@ You can use the buttons in the top-right corner to:
     This is a current limitation of the GDScript debugger.
 
 Errors
-++++++
+------
 
 This is where error and warning messages are printed while running the game.
 
 You can disable specific warnings in **Project Settings > Debug > GDScript**.
 
 Profiler
-++++++++
+--------
 
 The profiler is used to see what code is running while your project is in use,
-and how that effects performance. A detailed explanation of how to use it can
-be found :ref:`here <doc_the_profiler>`.
+and how that effects performance.
+
+.. seealso::
+
+    A detailed explanation of how to use the profiler can be found in the
+    dedicated :ref:`doc_the_profiler` page.
+
+Visual Profiler
+---------------
+
+The Visual Profiler can be used to monitor what is taking the most time when
+rendering a frame on the CPU and GPU respectively. This allows tracking sources
+of potential CPU and GPU bottlenecks caused by rendering.
+
+.. warning::
+
+    The Visual Profiler only measures CPU time taken for rendering tasks, such
+    as performing draw calls. The Visual Profiler does **not** include CPU time
+    taken for other tasks such as scripting and physics. Use the standard
+    Profiler tab to track non-rendering-related CPU tasks.
+
+To use the visual profiler, run the project, switch to the **Visual Profiler**
+tab within the Debugger bottom panel, then click **Start**:
+
+.. figure:: img/debugger_visual_profiler_results.webp
+   :alt: Visual Profiler tab after clicking Start, waiting for a few seconds, then clicking Stop
+
+   Visual Profiler tab after clicking **Start**, waiting for a few seconds, then clicking **Stop**
+
+You will see categories and results appearing as the profiler is running. Graph
+lines also appear, with the left side being a CPU framegraph and the right side
+being a GPU framegraph.
+
+Click **Stop** to finish profiling, which will keep the results visible but
+frozen in place. Results remain visible after stopping the running project, but
+not after exiting the editor.
+
+Click on result categories on the left to highlight them in the CPU and GPU
+graphs on the right. You can also click on the graph to move the cursor to a
+specific frame number and highlight the selected data type in the result
+categories on the left.
+
+You can switch the result display between a time value (in milliseconds per
+frame) or a percentage of the target frametime (which is currently hardcoded to
+16.67 milliseconds, or 60 FPS).
+
+If framerate spikes occur during profiling, this can cause the graph to be
+poorly scaled. Disable **Fit to Frame** so that the graph will zoom onto the 60
+FPS+ portion.
+
+.. note::
+
+    Remember that Visual Profiler results will vary **heavily** based on
```

Comment:
```suggestion
    Remember that Visual Profiler results can vary **heavily** based on
```

This won't be the case for all users. It depends on where their bottlenecks are. If the game is CPU-bottlenecked window size won't affect performance. Similarly if the GPU bottleneck is a memory load bottleneck in the vertex shader, then changing window size may not affect performance either. 

But, the advice is good in most situations as Godot tends to hit fill-rate bottlenecks pretty quickly. So it is worth keeping this note in place, we should just soften the language a little bit